cs/test/core/JsonParsingTests.cs

```diff
@@ -703,5 +703,25 @@ static T ParseJson<T>(string json) where T : new()
             var target = deserializer.Deserialize<T>(reader);
             return target;
         }
+
+        [Test]
+        public void JsonParsing_DateAsStringInStreamConstructor()
+        {
+            string timestamp = "2017-03-07T04:47:50.1145227Z";
+            string json = $"{{\"timestamp\":\"{timestamp}\"}}";
+
+            DateAsString target;
+            using (MemoryStream ms = new MemoryStream())
```

Comment:
How about the following instead?

```
var ms = new MemoryStream(Encoding.UTF8.GetBytes(json));
var reader = new SimpleJsonReader(ms);
var target = Deserialize<DataAsString>.From(reader);
Assert.AreEqual(timestamp, target.timestamp);
```

We can also skip disposing the MemoryStream. MemoryStream.Dispose doesn't do anything interesting.